boards/arm/mimxrt595_evk/board.c

```diff
@@ -93,7 +119,148 @@ static int board_config_pmic(void)
 
 	return ret;
 }
-#endif
+
+static int board_pmic_change_mode(uint8_t pmic_mode)
+{
+	int ret;
+
+	if (pmic_mode >= 4) {
+		return -ERANGE;
+	}
+
+	ret = regulator_parent_dvs_state_set(pca9420, pmic_mode);
+	if (ret != -EPERM) {
+		return ret;
+	}
+
+	POWER_SetPmicMode(pmic_mode, kCfg_Run);
```

Comment:
Hi @decsny ,
There is no Zephyr API I am aware of to do this.  [POWER_SetPmicMode()](https://github.com/zephyrproject-rtos/hal_nxp/blob/2a294b540c09b36f7cddece44d25628bfde5970e/mcux/mcux-sdk/devices/MIMXRT595S/drivers/fsl_power.c#L1275) is not an API for the PMIC itself.  Instead, the RT500 MCU has 2 pins that signal an external PMIC, called the PMIC_MODE pins.  This allows the PMIC to have 4 pre-configured modes.  Calling `POWER_SetPmicMode()` here changes the output on the PMIC_MODE pins to have the PMIC switch modes.

Do you think there is a better option?

Reply:
I am not opposed to what you have here. I was just wondering should we have a regulator driver for this? the regulator subsystem includes drivers for very simple things like even a single gpio that just toggles an external transistor. I am wondering if it could be a good pattern to follow, since I would imagine in the future, a generic power framework would make use of the regulator subsystem in some way to do things like dvfs. But in this case, it looks like this function makes use of SOC specific sysctl registers. And zephyr does not have a generic framework that can do dvfs. So I am not sure if this would be worth the effort.